### PR TITLE
Several fixes + working on simple controller

### PR DESCRIPTION
**Pull request by mjh-sakh**

### CODE REVIEW

File: Stage1/environment/env_1d.py [hunk 1]
```diff
@@ -2,7 +2,37 @@
 import cv2
 import os
 import gym
+import pyglet
 from gym import spaces
+from gym.envs.classic_control import rendering
+
+
+class ImageObject:
+    def __init__(self, bites):
```
Comment (olegboev) [inline Stage1/environment/env_1d.py]: Class and method comments are missing

File: Stage1/environment/env_1d.py [hunk 2]
```diff
@@ -2,7 +2,37 @@
...
+        self.bites = bites
```
Comment (olegboev) [inline Stage1/environment/env_1d.py]: Probably the correct name is "bytes"

File: Stage1/environment/env_1d.py [hunk 3]
```diff
@@ -2,7 +2,37 @@
...
+
+    def read(self):
+        return self.bites
+
+
+class NPImage(rendering.Geom):
```
Comment (olegboev) [inline Stage1/environment/env_1d.py:22]: Please use an informative class name. It is not clear what does "NP" mean here.

File: Stage1/environment/env_1d.py [hunk 4]
```diff
@@ -2,7 +2,37 @@
...
+    def __init__(self, fname, array_img, width, height):
```
Comment (olegboev) [inline Stage1/environment/env_1d.py:27]: Class and method comments are missing

File: Stage1/environment/env_1d.py [hunk 5]
```diff
@@ -20,7 +50,7 @@ def __init__(self, plan_file_path: str):
             0: speed, px/sec
```
Comment (olegboev) [inline Stage1/environment/env_1d.py:58]: What does "1d" in the name Env1d mean? I suggest naming this class as "Environment"

File: Stage1/environment/env_1d.py [hunk 6]
```diff
@@ -20,7 +50,7 @@ def __init__(self, plan_file_path: str):
...
             1: turn, radians 
         """
-        self.action_space = spaces.Box(low=[0, -2 * np.pi], high=[100, 2 * np.pi])
```
Comment (olegboev) [inline Stage1/environment/env_1d.py:23]: Please add the comments describing what is going on here, and what these magic numbers are. In general, magic numbers should be changed to named constants or described in the comments.

File: Stage1/main.py [hunk 7]
```diff
@@ -34,14 +35,40 @@ def main():
 
         image_output = draw_pland_and_frames(plan_with_scan, frame_prev, frame_curr, matches)
 
-demo_map = Env1d(r'environment/demo_map_1.jpg')
 
-for _ in range(20):
-    scan = camera_1d(demo_map.plan, demo_map.drone_coordinates, np.random.rand() * 2 * np.pi, np.pi / 6, 20, is_wall)
-    demo_map.drone_coordinates = demo_map.place_drone()
+def simple_work_with_env():
```
Comment (olegboev) [inline Stage1/main.py]: Please use the informative function name instead of "simple_work_with_env()"

File: Stage1/environment/env_1d.py [hunk 8]
```diff
@@ -61,30 +91,48 @@ def reset(self):
...
         if self.viewer is None:
             from gym.envs.classic_control import rendering
             self.viewer = rendering.Viewer(screen_width, screen_height)
-
-        # TODO: add plan
-        # gym doesn't work with images, only primitive forms, so need to find a way to draw a np.array
-        # np.array can be generated by show_layers
+            plan = NPImage(self.plan_file_path, self.plan, screen_width, screen_height)
+            self.viewer.add_geom(plan)
+            drone = rendering.FilledPolygon([(-20, -5), (-20, 5), (0, 0)])
```
Comment (olegboev) [inline Stage1/environment/env_1d.py:111]: Please use named constants instead of magic numbers.

File: Stage1/environment/env_1d.py [hunk 9]
```diff
@@ -2,7 +2,37 @@
 import cv2
 import os
 import gym
+import pyglet
 from gym import spaces
+from gym.envs.classic_control import rendering
+
+
+class ImageObject:
```
Comment (olegboev) [inline Stage1/environment/env_1d.py:10]: Please use informative class names.  How does "ImageObject" provide any information on the class purpose or functionality?

File: Stage1/environment/env_1d.py [hunk 10]
```diff
@@ -2,7 +2,37 @@
...
+        self.height = height
+        file = self.conv_array_to_bytes(array_img)
+        img = pyglet.image.load(fname, file=file)
+        self.img = img
+        self.flip = False
+
+    def render1(self):
+        self.img.blit(0, 0, width=self.width, height=self.height)
+
+    def conv_array_to_bytes(self, array: np.array):
```
Comment (olegboev) [inline Stage1/environment/env_1d.py]: Please use clear names. What does "conv" mean? If it is "convert", then name it "convert_...".

File: Stage1/environment/env_1d.py [hunk 11]
```diff
@@ -61,30 +91,48 @@ def reset(self):
...
-        # TODO: add plan
-        # gym doesn't work with images, only primitive forms, so need to find a way to draw a np.array
-        # np.array can be generated by show_layers
+            plan = NPImage(self.plan_file_path, self.plan, screen_width, screen_height)
+            self.viewer.add_geom(plan)
+            drone = rendering.FilledPolygon([(-20, -5), (-20, 5), (0, 0)])
+            self.dronetrans = rendering.Transform()
+            drone.add_attr(self.dronetrans)
+            self.viewer.add_geom(drone)
+
```
Comment (olegboev) [inline Stage1/environment/env_1d.py:115]: Please provide comments to the method body in order to simplify code understanding.

**Review by olegboev — CHANGES_REQUESTED** [review]

File: Stage1/environment/env_1d.py (hunk 5 above)
Comment (mjh-sakh) [reply Stage1/environment/env_1d.py:58]: This comment is not for this PR. 1d was there before. Agree with naming. Please create issue for refactor.

File: Stage1/environment/env_1d.py (hunk 8 above)
Comment (mjh-sakh) [reply Stage1/environment/env_1d.py:111]: It is drawing arrow by providing 3 vertices coordinates. I don't see need for naming those ones. Please approve and let continue discussion in slack.

File: Stage1/environment/env_1d.py (hunk 9 above)
Comment (mjh-sakh) [reply Stage1/environment/env_1d.py:10]: Addressed with class description. Name is super appropriate :)

File: Stage1/environment/env_1d.py (hunk 11 above)
Comment (mjh-sakh) [reply Stage1/environment/env_1d.py:115]: I will muddy the code by comments as OpenAI Gym mechanics understanding are required. I don't want to describe all iceberg where only tip of the iceberg is shown. 
I will explain how OpenAI Gym works for people in slack.

**Pull request merged**